src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.UserDao;
+import com.dev.cinema.exception.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import java.util.Optional;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Fail! Can't add user: " + user, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query =
+                    session.createQuery("FROM User WHERE email = :userEmail ", User.class);
+            query.setParameter("userEmail", email);
+            return Optional.of(query.getSingleResult());
+        } catch (Exception e) {
+            throw new DataProcessingException(
+                    "User couldn't found, by email: " + email + " ", e);
+        }
```

Comment:
No need to use the catch block since we return the **_Optional_**
If we don't find the required user by such an email we return **_Optional.empty()_**
Then if we try to extract the value and there is no such found we receive a **_NullPointerException_**
That's why this handling makes no sense because you simply won't receive an exception in here

Reply:
agree